pkg/apiserver/registry/system/bundle/dump.go

```diff
@@ -0,0 +1,194 @@
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bundle
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"crypto/sha256"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/spf13/afero"
+)
+
+func dumpFlows(basedir string) error {
+	brListOutput, err := defaultExecutor.Command("ovs-vsctl", "list-br").CombinedOutput()
```

Comment:
this will be in antrea-controller?

Reply:
These functions would be picked up depends on components. see bundle/rest.go:212 and bundle/rest.go:227.

Reply:
I also feel Agent specific func should be put in pkg/agent/ (how about pkg/agent/apiserver/handlers/)?
Or at least implementation of dumping flows, bridges, iptables should be in pkg/agent/.

@tnqn : what is your opinion what should be in pkg/apiserver, what should be in pkg/controller or agent/?

Reply:
For dumping flows with ofctl, it is already wrapped in agent/pkg/ovsctl. I think you should add ovs-vsctl funcs there too.

Reply:
Since the REST is used for both agent and controller,  I think it is better to locate at the current position.
But you'd like to expose these dump functions as APIs, it would be good to be placed under agent directory.

Reply:
I was thinking about two ways:
1) expose two funcs from controller and agent respectively that dump all state to the passed dir. apisever/apis/system/bundle/ just calls these two funcs.
2) move implementation of controller/agent specific logics like bridge dumping, iptables dumping to controller/agent, and but keep the callers in apisever/apis/system/bundle/.

In general, I would not expose too much controller/agent internal logic/state in apiserver/.

Reply:
Updated, PTAL.